### ISSUE TITLE
Features Selected by SULOV depend on the version of featurewiz

### ISSUE THREAD

**Issue by epris** [AutoViML/featurewiz#42]
Dear all,
I was using featurewiz version 0.0.38 in an old project (https://github.com/AutoViML/featurewiz/tree/6b870dae8dcf4f24873eb61bb48947ceb84e189c)
The number of features selected and returned by FE_remove_variables_using_SULOV_method was 18

I am using featurewiz version 0.1.87 in a new project 
The number of features selected and returned  by FE_remove_variables_using_SULOV_method is 43

The dataset in input and the inputs parameters to featurewiz are the same for both projects. I executed step by step the two versions of featurewiz and I could say that the results of the two versions diverge starting from the computation of the correlation matrix at the beginning of FE_remove_variables_using_SULOV_method. Moreover, I noticed that the target label is modified by                     mlb = My_LabelEncoder() and dataname[each_target] = mlb.fit_transform(dataname[each_target]) in the new version before calling SULOV but it didn't happen in the previous version. 

Could you clarify what main differences have been introduced in the new version? As you can imagine, such a significant difference in the outputs of the two versions is unpleasant.

> Imported featurewiz: advanced feature engg and selection library. Version=0.0.38
> output = featurewiz(dataname, target, corr_limit=0.70,
>                     verbose=2, sep=',', header=0, test_data='',
>                     feature_engg='', category_encoders='')
> Create new features via 'feature_engg' flag : ['interactions','groupby','target']
>                                 
> Skipping feature engineering since no feature_engg input...
> Skipping category encoding since no category encoders specified in input...
> Shape of your Data Set loaded: (38, 3385)
>     Filename is an empty string or file not able to be loaded
> ############## C L A S S I F Y I N G  V A R I A B L E S  ####################
> Classifying variables in data set...
>     3384 Predictors classified...
> 2022-07-26 00:03:54.391147: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
> To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
>         142 variable(s) will be ignored since they are ID or low-information variables
> Shape of your Data Set loaded: (38, 3385)
> Number of processors on machine = 1
>     No GPU active on this device
>     Running XGBoost using CPU parameters
> ############## C L A S S I F Y I N G  V A R I A B L E S  ####################
> Classifying variables in data set...
>     3384 Predictors classified...
>         142 variable(s) will be ignored since they are ID or low-information variables
> Removing 142 columns from further processing since ID or low information variables
>     columns removed: ['x427', 'x433', 'x439', 'x771', 'x777', 'x783', 'x825', 'x831', 'x837', 'x850', 'x856', 'x862', 'x1216', 'x1228', 'x1240', 'x1248', 'x1254', 'x1260', 'x1273', 'x1279', 'x1285', 'x1289', 'x1301', 'x1313', 'x1617', 'x1623', 'x1629', 'x1633', 'x1639', 'x1645', 'x1651', 'x1657', 'x1663', 'x1671', 'x1677', 'x1683', 'x1696', 'x1702', 'x1708', 'x1712', 'x1724', 'x1736', 'x2056', 'x2062', 'x2068', 'x2074', 'x2080', 'x2086', 'x2094', 'x2100', 'x2106', 'x2119', 'x2125', 'x2131', 'x2135', 'x2147', 'x2159', 'x2479', 'x2491', 'x2503', 'x2517', 'x2523', 'x2529', 'x2542', 'x2548', 'x2554', 'x2558', 'x2570', 'x2582', 'x2965', 'x2971', 'x2977', 'x2981', 'x2993', 'x3005', 'x3325', 'x3337', 'x3349', 'x3363', 'x3369', 'x3375', 'x490', 'x495', 'x497', 'x502', 'x503', 'x507', 'x844', 'x913', 'x918', 'x920', 'x924', 'x925', 'x926', 'x936', 'x1267', 'x1268', 'x1330', 'x1332', 'x1334', 'x1336', 'x1341', 'x1343', 'x1347', 'x1348', 'x1349', 'x1351', 'x1360', 'x1361', 'x1753', 'x1755', 'x1757', 'x1759', 'x1764', 'x1766', 'x1770', 'x1771', 'x1772', 'x1778', 'x2176', 'x2178', 'x2180', 'x2182', 'x2187', 'x2189', 'x2194', 'x2195', 'x2207', 'x2599', 'x2601', 'x2603', 'x2605', 'x2610', 'x2616', 'x2617', 'x2959', 'x3022', 'x3024', 'x3026', 'x3028', 'x3039', 'x3046']
>     After removing redundant variables from further processing, features left = 3242
> #### Single_Label Binary_Classification Feature Selection Started ####
> Searching for highly correlated variables from 3242 variables using SULOV method
> #####  SULOV : Searching for Uncorrelated List Of Variables (takes time...) ############
>     Removing (3224) highly correlated variables:
>     Following (18) vars selected: ['x2', 'x10', 'x26', 'x69', 'x87', 'x129', 'x187', 'x417', 'x496', 'x554', 'x608', 'x975', 'x1033', 'x1156', 'x1765', 'x2134', 'x2910', 'x3176']

> Imported version = 0.1.87.
> from featurewiz import FeatureWiz
> wiz = FeatureWiz(verbose=1)
> X_train_selected = wiz.fit_transform(X_train, y_train)
> X_test_selected = wiz.transform(X_test)
> wiz.features  ### provides a list of selected features ###
>                                 
> ############################################################################################
> ############       F A S T   F E A T U R E  E N G G    A N D    S E L E C T I O N ! ########
> # Be judicious with featurewiz. Don't use it to create too many un-interpretable features! #
> ############################################################################################
> Skipping feature engineering since no feature_engg input...
> Skipping category encoding since no category encoders specified in input...
> **INFO: featurewiz can now read feather formatted files. Loading train data...
>     Shape of your Data Set loaded: (38, 3385)
>     Loaded train data. Shape = (38, 3385)
> No test data filename given...
> #######################################################################################
> ######################## C L A S S I F Y I N G  V A R I A B L E S  ####################
> #######################################################################################
> Classifying variables in data set...
>     3384 Predictors classified...
>         142 variable(s) to be removed since ID or low-information variables
>     	more than 142 variables to be removed; too many to print...
> train data shape before dropping 81 columns = (38, 3385)
> 	train data shape after dropping columns = (38, 3304)
>     Converted pandas dataframe into a Dask dataframe ...
> No GPU active on this device
>     Tuning XGBoost using CPU hyper-parameters. This will take time...
>     After removing redundant variables from further processing, features left = 3242
> No interactions created for categorical vars since feature engg does not specify it
> #### Single_Label Binary_Classification problem ####
>     target labels need to be converted...
> Completed label encoding of target variable = target
> How model predictions need to be transformed for target:
> 	{0: 1}
> #######################################################################################
> #####  Searching for Uncorrelated List Of Variables (SULOV) in 3242 features ############
> #######################################################################################
>     there are no null values in dataset...
>     Removing (3199) highly correlated variables:
>     SULOV method is erroring. Continuing ...
> Time taken for SULOV method = 138 seconds
>     Adding 0 categorical variables to reduced numeric variables  of 3242
> Final list of selected vars after SULOV = 3242
> Readying dataset for Recursive XGBoost by converting all features to numeric...

**Comment by AutoViML** [comment]
Hi @epris 👍 
The problem with the second time is seen in your above issue:
```
SULOV method is erroring. Continuing ...
```
Since SULOV method is erroring, it gives a different result. Can you please upgrade to the latest version via:
```
pip install featurewiz --upgrade
```

Then try again and see if it is indeed that different. When there are 3000+ features while there are only 38 rows, you can expect some variance in results. Just FYI.

Autovimal

**Comment by epris** [comment]
Hi,
thanks for your response.
Well, I know that 38 samples are few but given the same input, given the same seed, if the algorithm is the same the output is the same. 
But, as you noticed, there is a problem with the installation.
Unfortunately, I run the command you suggested but the upgrade get stuck at:
INFO: pip is looking at multiple versions of cloudpickle to determine which version is compatible with other requirements. This could take a while.
Collecting cloudpickle>=1.1.1
  Using cached cloudpickle-2.0.0-py3-none-any.whl (25 kB)
  Downloading cloudpickle-1.6.0-py3-none-any.whl (23 kB)
  Using cached cloudpickle-1.5.0-py3-none-any.whl (22 kB)
  Using cached cloudpickle-1.4.1-py3-none-any.whl (26 kB)
  Using cached cloudpickle-1.4.0-py3-none-any.whl (25 kB)
  Using cached cloudpickle-1.3.0-py2.py3-none-any.whl (26 kB)
  Using cached cloudpickle-1.2.2-py2.py3-none-any.whl (25 kB)
  Using cached cloudpickle-1.2.1-py2.py3-none-any.whl (25 kB)
  Using cached cloudpickle-1.2.0-py2.py3-none-any.whl (24 kB)
  Using cached cloudpickle-1.1.1-py2.py3-none-any.whl (17 kB)
INFO: pip is looking at multiple versions of cloudpickle to determine which version is compatible with other requirements. This could take a while.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking

**Comment by AutoViML** [comment]
Hi @epris 👍 
Thanks for trying out Featurewiz. Here is a Colab notebook which shows how to install featurewiz by carefully following these steps one by one:

```
!pip install featurewiz --ignore-installed --no-deps
!pip install 'fsspec>=0.3.3'
!pip install category_encoders --upgrade
!pip install imbalanced_ensemble 
```
https://colab.research.google.com/drive/1ekBZgCDgRGJU25unwZ0lvla_k20pq36h?usp=sharing
